src/__test__/connection/connectionStringMockups.ts

```diff
@@ -415,7 +470,7 @@ export const valid: Array<
     },
   ],
   [
-    "esdb://host?maxDiscoverAttempts=200&discoveryInterval=1000&gossipTimeout=1&nodePreference=leader&tls=false&tlsVerifyCert=false&throwOnAppendFailure=false",
+    "esdb://host?maxDiscoverAttempts=200&discoveryInterval=1000&gossipTimeout=1&nodePreference=leader&tls=false&tlsVerifyCert=false&throwOnAppendFailure=false&keepAliveInterval=10",
```

Comment:
**SUGGESTION:** Can we add also the scenario for `keepAliveTimeout`?

Reply:
> Can we add also the scenario for non-default keepAliveTimeout

See [line 494](https://github.com/EventStore/EventStore-Client-NodeJS/pull/135/files#diff-513c77abcbb34a10b9a67df5be0345a17d27aebab9b31a0e34e34c65ab7f1a58R494)

> It would be also good to verify how the code behaves permutations if e.g. keepAliveInterval is equal to -1 and keepAliveTimeout is non default, etc.

These tests only handle the parsing, and each argument is handled individually, without context so there is no value in checking permutations of two related arguement.

Reply:
Tests should not have any assumptions about the implementation. I know that the current implementation does not connect those cases, but implementation can change, and black-box testing should catch the behaviour change. If those tests do not verify if the settings are applied, then we should add such.

Reply:
>  I know that the current implementation does not connect those cases, but implementation can change, and black-box testing should catch the behaviour change.
Sure, but within reason. You could equally argue that the implementation could change that checks their position or ordering in the connection string. Currently there is no requirement for the arguments to be related, so if that gets added, the test gets added. 

> If those tests do not verify if the settings are applied, then we should add such.
Yeah, I'll add some tests for the error and warning cases when I add the warning.

Reply:
Just to make sure we are on the same page, you realise these are part of the unit test of the internal connection string parsing function, not test cases for different configurations of the client?

Reply:
I'm okay with that, as long those cases are covered somewhere.

Reply:
> Yeah, I'll add some tests for the error and warning cases when I add the warning.

added here: 

https://github.com/EventStore/EventStore-Client-NodeJS/pull/135/files#diff-187f6e28b178ac68395525a741af0ccade761fde1e0280c1eaba08f00341e089